### ISSUE TITLE
Retrieve localized metazone info from timeZoneNames.json in TimeZonesProvider

### ISSUE THREAD

**Issue by nordzilla** [unicode-org/icu4x#993]
We currently pull the general metazone localizations from timeZonNames.json, but some time zones have special localizations for their metazones, such as "British Summer Time" for the London time zone. 

We need to add this information to the TimeZonesProvider and utilize it when formatting.

**Comment by nordzilla** [comment]
@samchen61661 

### Background Info

```
icu4x
├── components
├── docs
├── experimental
├── ffi
├── proposals
├── provider
├── target
├── tools
└── utils
```

In ICU4X we maintain two sets of data types: one set in the `components` directory, and one set in the `provider` directory. 

---

The types in the `components` directory are structured in a way that is convenient for us to work with in our library for formatting etc. For time zones, let's look at the `TimeZoneFormatsV1` type on the `components` side:
> [`icu4x/components/datetime/src/provider/time_zones.rs#L52-L66`](https://github.com/unicode-org/icu4x/blob/608fbd45586a8f392421b30087c6cd1c2cf902ca/components/datetime/src/provider/time_zones.rs#L52-L66)

---

The types in the `provider` directory are structured in a way that is convenient to deserialize from the CLDR JSON data (similar to how we set up our test types with the JSON fixtures). 

You can see how the `TimeZoneNames` struct
> [`icu4x/provider/cldr/src/transform/time_zones/cldr_json/mod.rs#L76-L86`](https://github.com/unicode-org/icu4x/blob/608fbd45586a8f392421b30087c6cd1c2cf902ca/provider/cldr/src/transform/time_zones/cldr_json/mod.rs#L76-L86)

closely matches the CLDR JSON structure 
> [`cldr-json/cldr-dates-full/main/en/timeZoneNames.json#L11-L18`](https://github.com/unicode-org/cldr-json/blob/83f6144d8aa3d10e932c25924d4c903925b24257/cldr-json/cldr-dates-full/main/en/timeZoneNames.json#L11-L18)

---

Then, we define conversions between the JSON-like Rust types on the `provider` side, into the preferred ICU4X data representation on the `components` side using the [`From`](https://doc.rust-lang.org/std/convert/trait.From.html) trait, just like you did for updating the test fixtures in #1256.
> [`icu4x/provider/cldr/src/transform/time_zones/cldr_json/convert.rs#L35-L58`](https://github.com/unicode-org/icu4x/blob/608fbd45586a8f392421b30087c6cd1c2cf902ca/provider/cldr/src/transform/time_zones/cldr_json/convert.rs#L35-L58)

---

The `timeZoneNames.json` file defines localized translations for the metazone names:
> [`cldr-json/cldr-dates-full/main/en/timeZoneNames.json#L1348-L1360`](https://github.com/unicode-org/cldr-json/blob/83f6144d8aa3d10e932c25924d4c903925b24257/cldr-json/cldr-dates-full/main/en/timeZoneNames.json#L1348-L1360)

And `timeZoneNames.json` also defines localized translations for the time zone names:
> [`cldr-json/cldr-dates-full/main/en/timeZoneNames.json#L19-L26`](https://github.com/unicode-org/cldr-json/blob/83f6144d8aa3d10e932c25924d4c903925b24257/cldr-json/cldr-dates-full/main/en/timeZoneNames.json#L19-L26)

Most of the entries for the time zones only have an `exemplarCity`, but occasionally they have some more specific metazone information, such as London prefers "British Summer Time" during daylight savings time:

> [`cldr-json/cldr-dates-full/main/en/timeZoneNames.json#L583-L587`](https://github.com/unicode-org/cldr-json/blob/83f6144d8aa3d10e932c25924d4c903925b24257/cldr-json/cldr-dates-full/main/en/timeZoneNames.json#L583-L587)

---

### What to do in this PR

We already read in this data from the JSON on the `provider` side here:
> [`icu4x/provider/cldr/src/transform/time_zones/cldr_json/mod.rs#L30-L52`](https://github.com/unicode-org/icu4x/blob/608fbd45586a8f392421b30087c6cd1c2cf902ca/provider/cldr/src/transform/time_zones/cldr_json/mod.rs#L30-L52)

But we currently only handle the exemplar city in the `components` side. 

In this PR, we would need to make a new `V1` type on the `components` side that carries this special-case data if the time zone ID has a more specific name for the metazone (like British Summer Time).

Next we would need to add a new conversion from the `provider` side to the `components` side for this new type.

Then we would need to use the new type in the formatting code, to check for the existence of a more-specific metazone translation, given the timezone ID, before we go with the general metazone translation. 

And we also need to add more test cases to ensure that this behavior is covered, of course.

**Comment by nordzilla** [comment]
Fixed by #1354 1354
